### DISCUSSION TITLE
Sonoff Dual R3 for Shutter

### DISCUSSION THREAD

**Discussion by RepiZoli** [arendst/Tasmota#17917] [Q&A] [answered]
Hi  

I new here, I have a Sonoff dual r3 what is flashed for tasmota: 
https://templates.blakadder.com/sonoff_DUALR3.html
However the result is that the two relay is working independent.--> For the shutter motor is not very healty 
I have the parameter template:
{"NAME":"Sonoff Dual R3","GPIO":[32,0,0,0,0,0,0,0,0,576,225,0,0,0,0,0,0,0,0,0,0,7296,7328,224,0,0,0,0,160,161,0,0,0,0,0,0],"FLAG":0,"BASE":1}
Anybody has a parameter template for shutter
Somthing like this?

<img width="265" alt="Capture" src="https://user-images.githubusercontent.com/99737125/217525560-5c351444-c848-40ac-9272-aad273bfbf14.PNG">

Thanks in advance
Or I have to adjust something else?

Thanks Zoli

**Comment by Jason2866** [comment, marked as answer]
Yes, you missed `interlock` All explained here https://tasmota.github.io/docs/Blinds-and-Shutters/

**Comment by dBmsquelch** [comment]
Hi!
Help me please! I'm very frustrated with this topic. When I set the shutter mode with ShutterRelay 1 on my SONOFF DUAL R3 I always get two shutters with a wrong logic:
![image](https://github.com/arendst/Tasmota/assets/76177868/4e06ad87-49ba-4cf3-b580-dd591b251641)

Is it a bug or my fault?
Thanks!

Edit:
I somehow managed to make the first shutter working but how can I disable the second one?

**Reply by Mark-Muc** [reply]
Hello . I also use the Sonoff R3 as a Shutter(driver),
I would advice you  doing like this.
Reset the Module to it's Defaults.
then enter in the console: (Shutter define):
`backlog Interlock 1,2; Interlock ON; SetOption80 1;Shuttermode 1; ShutterRelay1 1`
then 
define moving times for example Window; set your own times) 
`backlog ShutterOpenDuration1 22; ShutterCloseDuration1 21.5; ShutterSetHalfway1 65; ShutterEnableEndStopTime1`

that's it and it should work!!
Have fun
Mark_Muc

**Comment by dBmsquelch** [comment]
Thank you!
This worked for me. There are still two Shutters visible at the tasmota
main view but I can live with that...

Mark-Muc ***@***.***> schrieb am Do., 27. Juni 2024, 11:07:

**Reply by Mark-Muc** [reply]
I use this Template:
` {"NAME":"Sonoff Dual R3 v2 (switch) an LCN HU","GPIO":[32,0,0,0,0,0,0,0,0,576,225,0,0,0,0,0,0,0,0,0,0,3200,8128,224,0,0,0,0,160,161,0,0,0,0,0,0],"FLAG":0,"BASE":1}`
and it looks like that:
![image](https://github.com/arendst/Tasmota/assets/67410847/45a79da1-de05-459b-ae70-4302f9254080)

Mark-Muc

**Comment by vinzeno** [comment]
Hello, I still can't activate the shutter function during compilation, I check the right boxes, I don't understand. The shutter function doesn't seem to be activated. I'm desperate, do you have a solution please.
Kind regards
![2024-07-26 11_26_24-Tasmota - Console](https://github.com/user-attachments/assets/97af4147-83d5-42cd-8244-023d08e80389)

**Reply by sfromis** [reply]
Did you read the clear message about you needing `Interlock` as explained in the docs, and also mentioned in the beginning in this thread?

**Comment by vinzeno** [comment]
Yes and I don't understand because I added before the compilation the #define USE_SHUTTER_ESP32.
I even tried with #define USE_SHUTTER because some forums say that it is this function that must be put before compiling.
I tried the Dev version and I also have the same result.
Thank you for your help.

**Reply by sfromis** [reply]
Well, you need to configure Interlock, like using the `Interlock` command. That's what the message tells you. This depends on what shutter mode you're using, but as the [docs](https://tasmota.github.io/docs/Blinds-and-Shutters/#shutter-modes) say:
> At least for Shutter mode 1 an interlock is mandatory

There's also an example of using it with mode 1.

**Reply by vinzeno** [reply]
Thank you very much, in fact I had not understood that it was necessary to put a relay on for the Interlock command to work.
It's perfect, thank you very much, everything works
